### PR TITLE
Adds support for Elasticsearch 7.10.0

### PR DESCRIPTION
**Pull request by bowenlan-amzn**
*Description of changes:*
1. integTest change [PR](https://github.com/elastic/elasticsearch/pull/61879) [example](https://github.com/elastic/elasticsearch/pull/61879/files#diff-6f1b376b5c089cbea9807a58266d657e627c9220fbe341316cbfa2818cb02567R3)
since their javaRestTest create its own sourceSet and binding it to their test env, we choose to create a task (RestIntegTestTask), still name as `integTest`. In the back, same name cluster will be created, so don't need to worry about cluster creation. But we need to bundle our plugin into the cluster.
   - old [RestIntegTestTask](https://github.com/elastic/elasticsearch/blob/604c6dd528b10ae28ebf2c56ca1c08b5516de807/buildSrc/src/main/java/org/elasticsearch/gradle/test/RestIntegTestTask.java#L46)
   - example to write own RestIntegTestTask [code link](https://github.com/elastic/elasticsearch/blob/f6b3148e5e3758af3cf48fcb3cce9b55589c778b/x-pack/plugin/security/qa/basic-enable-security/build.gradle#L25)

2. No executor anymore in localmasterlistener [PR](https://github.com/elastic/elasticsearch/commit/c8ef2e18f753b09625142fb8816a7d0ec77f8a30#diff-68a8b0d752bfdf38bccb82c3f1595b6a80780cbce20094a369d78c37d9da6f4a)

3. `run` task now need to explicitly set cluster

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/alerts/AlertIndices.kt [hunk 1]
```diff
@@ -147,15 +147,28 @@ class AlertIndices(
...
-    override fun executorName(): String {
+    private fun executorName(): String {
         return ThreadPool.Names.MANAGEMENT
     }
 
     override fun clusterChanged(event: ClusterChangedEvent) {
+        // Instead of using a LocalNodeMasterListener to track master changes, this service will
+        // track them here to avoid conditions where master listener events run after other
+        // listeners that depend on what happened in the master listener
+        val prevIsMaster = this.isMaster
```
Comment (lezzago) [inline alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/alerts/AlertIndices.kt:162]: no need for this additional variable

**Comment by dbbaughe** [comment]
Can you confirm the previous functionality still works.
i.e. can you still do all of these things:
Debug the source code with breakpoints
Debug the integration tests running against the integTest cluster
Run multi node tests (Did alerting have this? I don't see a multi node check on this PR)

File: README.md [hunk 2]
```diff
@@ -84,18 +84,20 @@ OR
...
-The Elasticsearch server JVM will launch suspended and wait for a debugger to attach to `localhost:5005` before starting the Elasticsearch server.
-The IDE needs to listen for the remote JVM. If using Intellij you must set your debug configuration to "Listen to remote JVM" and make sure "Auto Restart" is checked.
-You must start your debugger to listen for remote JVM before running the commands.
+The Elasticsearch server JVM will launch suspended and wait for a debugger to attach to `localhost:5005` before starting the Elasticsearch server. The IDE needs to listen for the remote JVM. If using Intellij you must set your debug configuration to "Listen to remote JVM" and make sure "Auto Restart" is checked. You must start your debugger to listen for remote JVM before running the commands.
 
 To debug code running in an integ test (which exercises the server from a separate JVM), run:
 
 ```
-./gradlew -Dtest.debug :alerting:integTest 
+./gradlew :alerting:integTest-Dtest.debug
```
Comment (ylwu-amzn) [inline README.md:92]: Not space between `:alerting:integTest` and `-Dtest.debug`?

File: core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/schedule/JobSchedulerMetrics.kt [hunk 3]
```diff
@@ -48,7 +48,7 @@ class JobSchedulerMetrics : ToXContentFragment, Writeable {
 
     override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
         if (lastExecutionTime != null)
-            builder.timeField("last_execution_time", "last_execution_time", Instant.ofEpochMilli(lastExecutionTime).toEpochMilli())
+            builder.timeField("last_execution_time_millis", "last_execution_time", Instant.ofEpochMilli(lastExecutionTime).toEpochMilli())
```
Comment (lezzago) [inline core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/schedule/JobSchedulerMetrics.kt:51]: I thought we wanted to add `_in_millis` to the human readable parameter?

File: alerting/build.gradle [hunk 4]
```diff
@@ -88,61 +82,74 @@ dependencyLicenses.enabled = false
...
-                cluster.waitForAllConditions()
-            }
-        }
 
-        // The --debug-jvm command-line option makes the cluster debuggable; this makes the tests debuggable
-        if (System.getProperty("test.debug") != null) {
-            jvmArgs '-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005'
-        }
-    }
+import org.elasticsearch.gradle.test.RestIntegTestTask
```
Comment (lezzago) [inline alerting/build.gradle:96]: imports should be at the top

Comment (skkosuri-amzn) [reply alerting/build.gradle:96]: +1

File: core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/schedule/JobSchedulerMetrics.kt (hunk 3 above)
Comment (bowenlan-amzn) [reply core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/schedule/JobSchedulerMetrics.kt]: yes, fixed

File: alerting/build.gradle (hunk 4 above)
Comment (bowenlan-amzn) [reply alerting/build.gradle]: meant to be clear for this task, move to top now

**Review by skkosuri-amzn — APPROVED** [review]

**Review by lezzago — APPROVED** [review]

**Pull request merged**